### DISCUSSION TITLE
Delegated Voting

### DISCUSSION THREAD

**Discussion by ben2x4** [DA0-DA0/dao-contracts#372] [Ideas]
## Motivation
Voter fatigue is a big challenge for fast moving DAO's with lots of proposals. Delegated voting is a standard feature in the majority of existing DAO tooling enabling less active members to delegate their votes to more active members. 

## Implementation 
The cw20-staking was designed with delegated voting in mind. The delegated voting contracts can be hooked into the staking contract. There are two potential methods for designing the delegated voting contracts. 

### Single Delegate
This is the most simple delegated voting contract. Each address is allowed to delegate all their voting power to a single other address. 

#### Pros
- Simple
- No issues with staking and unstaking

#### Cons
- Potential concentration of voting power

### Multi-Delegate
This is a more complex contract where each address can delegate an arbitrary amount of voting power to an arbitrary number of address up to their total voting power. This contract presents some complexity with staking and unstaking. When you stake the voting power would not be automatically delegated to your current delegates. When you unstake you would have to undelegate first.

#### Pros
- Distribute voting power
- Engage with more available delegates

#### Cons
- Harder to build
- Complications with staking and unstaking

## Conclusion

In the future we could offer both options, but I think we should first focus on the contracts and UI on one option. My choice is the Multi-Delegate implementation as it seems to better align with DAO DAO mission to foster decentralized governance. Single-delegate voting will most likely concentrate voting power in the DAO. Multi-Delegate allows members to easily shift their delegations and engage with multiple delegatees. It is also a familiar model from our users as it is very similar to tendermint staking.

**Comment by elsehow** [comment]
I agree that we should focus on the multi-delegate model. We should enable the most fine-grained delegation possible, whether or not we choose to support it in the UI right away. Multi-delegate delegation can and should someday be supported through UI features (for example, "spread evenly across the following delegates:"),

**Comment by Callum-A** [comment]
Thinking on how this would look, would this be another `voting module` which instantiates or takes cw20 stake, queries the stake at a height and then has it's own state representing how the user has divided their votes.
